### PR TITLE
Retrieve market prices core - Closes #464

### LINKED ISSUE #464: Retrieve market prices (market-service)
### Description

Create internal method `market.prices`

### Motivation

Retrieval from the original service has to be possible through the Moleculer service.

### Acceptance Criteria

- The desired method returns market prices
- Functional tests are written

#### Methods

`market.prices`

This method allows market price retrieval and can be used by the gateway.

#### Method parameters

_(no parameters required)_

#### Payload

```
{
    "data": [
        {
            "code": "LSK_BTC",
            "from": "LSK",
            "rate": "0.00009027",
            "to": "BTC",
            "updateTimestamp": 1616939448,
            "sources": ["kraken"]
        },
        {
            "code": "LSK_CHF",
            "from": "LSK",
            "rate": "35.3452",
            "to": "CHF",
            "updateTimestamp": 1616939448,
            "sources": ["kraken", "exchangeratesapi.io"]
        }
    ],
    "meta": {
        "count": 1
    }
}
```

#### Exceptions

503 SERVICE_UNAVAILABLE

```
{
    error: true,
    message: "Service not available",
    status: "SERVICE_UNAVAILABLE"
}
```

Service not available happens when the data is not yet collected and the cache is empty.
It may also occur when all APIs are no longer available (redundancy failed), and the cache is expired. It might result with HTTP 503 Service Unavailable on the Gateway side.

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #464 

### How was it solved?
- [x] Added methods/controllers in `services/market`
- [x] Added placeholder for implementation and return mock data
- [x] Update `gateway` to use method name as per specs document
- [x] Added `docker-compose` file for market service
- [x] Added functional tests

### How was it tested?
- [x] Local
- [x] Jenkins

### CODE REVIEW

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find a couple of minor observations.

File: Jenkinsfile [hunk 1]
```diff
@@ -85,23 +85,24 @@ pipeline {
 				}
 			}
 		}
-		stage('Perform unit tests') {
+		stage('Run microservices') {
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply Jenkinsfile:88]: The reason to perform unit tests after all services started is to run the functional tests for market service which requires  services `services/market/tests/functional/market.prices.test.js` else test will always fail with error `market service not found`

``` const result = await broker.call('market.prices', {});```

[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply Jenkinsfile]: As we discussed, I added separate stage for functional tests in `Jenkins`.

File: services/market/tests/functional/market.prices.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            result.data.forEach(price => {
+                expect(price).toEqual(expect.objectContaining({
+                    code: expect.any(String),
+                    from: expect.any(String),
+                    rate: expect.any(String),
+                    to: expect.any(String),
+                    updateTimestamp: expect.any(Number),
+                    sources: expect.any(Array),
+
+                }));
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/market/tests/functional/market.prices.test.js]: Done!

File: services/market/tests/functional/market.prices.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const { ServiceBroker } = require('moleculer');
+
+const broker = new ServiceBroker({
+    transporter: 'redis://localhost:6379/0',
+    logLevel: 'info',
+    requestTimeout: 15 * 1000,
+    logger: console,
+});
+
+const marketPriceSchema = Joi.object({
```
Comment (sameersubudhi) [inline services/market/tests/functional/market.prices.test.js:28]:
```suggestion
const marketPriceItemSchema = Joi.object({
```

File: services/market/tests/functional/market.prices.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+describe('Test market prices', () => {
+    beforeAll(() => broker.start());
+    afterAll(() => broker.stop());
+
+    describe('Connect to client and retrieve market prices', () => {
+        it('call market.prices', async () => {
+            const result = await broker.call('market.prices', {});
+            expect(result.data.length).toBeGreaterThanOrEqual(1);
+            expect(result.data).toBeInstanceOf(Array);
+            result.data.forEach(price => marketPriceSchema.validate(price));
```
Comment (sameersubudhi) [inline services/market/tests/functional/market.prices.test.js:46]:
```suggestion
            result.data.forEach(price => marketPriceItemSchema.validate(price));
```

**Review by sameersubudhi — APPROVED** [review]
LGTM overall 👍  
However, please consider the following nomenclature recommendation.

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**